pkg/operator/controller/dns/controller.go

```diff
@@ -661,26 +660,30 @@ func getIbmDNSProvider(dnsConfig *configv1.DNS, creds *corev1.Secret, instanceID
 	}
 
 	providerCfg := ibm.Config{
-		APIKey:     string(creds.Data["ibmcloud_api_key"]),
-		InstanceID: instanceID,
-		Zones:      zones,
-		UserAgent:  userAgent,
+		APIKey:    string(creds.Data["ibmcloud_api_key"]),
+		Zones:     zones,
+		UserAgent: userAgent,
 	}
 
 	if isPublic {
+		providerCfg.InstanceID = instanceCRN
 		provider, err := ibmpublicdns.NewProvider(providerCfg)
 		if err != nil {
 			return nil, fmt.Errorf("failed to initialize IBM CIS DNS provider: %w", err)
 		}
 		log.Info("successfully initialized IBM CIS DNS provider")
 		return provider, nil
 	} else {
+		matches := ibm.IBMResourceCRNRegexp.FindStringSubmatch(instanceCRN)
+		if len(matches) <= 0 {
```

Comment:
Can this be just `== 0`?

Reply:
> Can this be just `== 0`?

On reflection, can we just check for `nil` as the Go doc says a return value of nil indicates no match.

```go
// FindStringSubmatch returns a slice of strings holding the text of the
// leftmost match of the regular expression in s and the matches, if any, of
// its subexpressions, as defined by the 'Submatch' description in the
// package comment.
// **A return value of nil indicates no match**.
```

Reply:
I have updated it to check against nil.

Reply:
I will take another look tomorrow -- ran out of time today.